### DISCUSSION TITLE
Someone please help to establish connectivity on Windows 11

### DISCUSSION THREAD

**Discussion by mrmate2** [markqvist/Reticulum#484] [Sideband]
I will be very thankfull if someone will assist me to set correct parameters in config file of sideband. Application does not see my wifi network. What is the correct way to enter ammendments in config file? Sorry for my English, I am from Ukraine.

**Comment by zirpu** [comment]
Are running Sideband on Android or Linux?

**Reply by zirpu** [reply]
oops. sorry, just realized you said Windows 11.  faragher's response is correct.

**Comment by faragher** [comment]
What are you trying to connect to? 

You said you're looking to connect over WiFi on Windows, so I'm going to assume you're not using a LoRa radio, and you're already connected to your WiFi.

If you're trying to connect to a server, you'll need tell Reticulum what server you want. You should have a config file in `C:\Users\<Your User Name>\.reticulum` 

The configuration should be straightforward, but I'll post an example below:

```
# This is the default Reticulum config file.
# You should probably edit it to include any additional,
# interfaces and settings you might need.

# Only the most basic options are included in this default
# configuration. To see a more verbose, and much longer,
# configuration example, you can run the command:
# rnsd --exampleconfig


[reticulum]
  
  # If you enable Transport, your system will route traffic
  # for other peers, pass announces and serve path requests.
  # This should only be done for systems that are suited to
  # act as transport nodes, ie. if they are stationary and
  # always-on. This directive is optional and can be removed
  # for brevity.
  
  enable_transport = False
  
  
  # By default, the first program to launch the Reticulum
  # Network Stack will create a shared instance, that other
  # programs can communicate with. Only the shared instance
  # opens all the configured interfaces directly, and other
  # local programs communicate with the shared instance over
  # a local socket. This is completely transparent to the
  # user, and should generally be turned on. This directive
  # is optional and can be removed for brevity.
  
  share_instance = Yes
  
  
  # If you want to run multiple *different* shared instances
  # on the same system, you will need to specify different
  # shared instance ports for each. The defaults are given
  # below, and again, these options can be left out if you
  # don't need them.
  
  shared_instance_port = 37428
  instance_control_port = 37429
  
  
  # You can configure Reticulum to panic and forcibly close
  # if an unrecoverable interface error occurs, such as the
  # hardware device for an interface disappearing. This is
  # an optional directive, and can be left out for brevity.
  # This behaviour is disabled by default.
  
  panic_on_interface_error = No


[logging]
  # Valid log levels are 0 through 7:
  #   0: Log only critical information
  #   1: Log errors and lower log levels
  #   2: Log warnings and lower log levels
  #   3: Log notices and lower log levels
  #   4: Log info and lower (this is the default)
  #   5: Verbose logging
  #   6: Debug logging
  #   7: Extreme logging
  
  loglevel = 4


# The interfaces section defines the physical and virtual
# interfaces Reticulum will use to communicate on. This
# section will contain examples for a variety of interface
# types. You can modify these or use them as a basis for
# your own config, or simply remove the unused ones.

[interfaces]
  
  # This interface enables communication with other
  # link-local Reticulum nodes over UDP. It does not
  # need any functional IP infrastructure like routers
  # or DHCP servers, but will require that at least link-
  # local IPv6 is enabled in your operating system, which
  # should be enabled by default in almost any OS. See
  # the Reticulum Manual for more configuration options.
  
  #[[Default Interface]]
   # type = AutoInterface
   # enabled = Yes   

  [[RNS Testnet Dublin]]
    type = TCPClientInterface
    enabled = yes
    target_host = dublin.connect.reticulum.network
    target_port = 4965
			
  [[Between the Borders Node Romeo]]
    type = TCPClientInterface
    enabled = yes
    target_host = reticulum.betweentheborders.com
    target_port = 4242
```

This is my Windows 10 configuration, so it should work for you (I removed the radio stuff). It will connect you to the Testnet through both the Dublin and BTB servers (located in the US). Since AutoInterface doesn't really work on Windows, so it's disabled.

If you have a custom server, you can add it in the [interfaces] section like so:

```
  [[<Your Server Name>]]
    type = TCPClientInterface
    enabled = yes
    target_host = <Your domain name or IP>
    target_port = 4242
```

The port can be changed, but 4242 is a safe choice. If it doesn't work, you can check your firewall.

If you want to run a server yourself (either connected to the testnet or privately) you can add the following to your Interfaces:

```
  [[<Your Server Name>]]
    type = TCPServerInterface
    interface_enabled = True
    listen_ip = 0.0.0.0
    listen_port = 4242
```

The name needs to be unique and the port needs to match. There are other options, but this'll get you running.

If none of this is what you wanted, let me know and I'll try to help.

**Reply by mrmate2** [reply]
I am trying to connect just to my wifi router, to be ablae to communicate with android phone wich is in same wifi network.
In mentioned above config file I have to specify network interface, because  Autointerface is not supported yet....
Ido not know how to write it correctly in config file.

**Reply by mrmate2** [reply]
my 2 android phones are communicating between each other in same network without issue. I just need to add 1 Win11 PC.

**Reply by mrmate2** [reply]
I am on a ship, have wifi netwirk, very often without internet. I just want to use this network to exchange messages with crew.
Some of them use Android, some Win11 PC.

**Reply by faragher** [reply]
This is an AutoInterface issue specific to the way Windows handles IPv6. 

You can add the TCPServer block in your Interface section, then add TCP connections in the Android versions that point to that computer's IP address and port.

**Reply by mrmate2** [reply]
Ok, that is clear. Thanks for advice.But what will be if mentioned PC will  be temporary not  connected, but another PC(laptop) will connect to same network? 
In android app is clear, every new device connected to same network has to announce himself, other users see him, and connect to him.
This is the way I did between smartphones. And in such way I can connect 3rd, 4th, 5th smartphone....
But how to do it in the network with smartphones and sometimes changing number of PCs.?

**Reply by mrmate2** [reply]
Do I need always to run 1 PC as server?  As I understand in android the only thing necessary-some number of smartphones connected to same wifi. In case of  Windows  client not same?

**Reply by faragher** [reply]
> I am on a ship, have wifi netwirk, very often without internet. I just want to use this network to exchange messages with crew. Some of them use Android, some Win11 PC.

Ah! Excellent. If you have a machine on the network as a server, you can have everything else connect to that one. The ones with AutoInterface can also see each other directly. So long as you have a single Windows, Mac, or Linux system configured with a server, then anything else can connect to it.

**Reply by mrmate2** [reply]
That is a problem, I do not have constant PC in network. I have common wifi network, 4-5 smartphones and sometimes 1-3 laptops.
No issues with smartphones, but is there away just to connect PC same way(without servers)? Just to connect PC to wifi and be able to see other smartphone....
Sorry for being annoying....

**Reply by faragher** [reply]
> Do I need always to run 1 PC as server? As I understand in android the only thing necessary-some number of smartphones connected to same wifi. In case of Windows client not same?

No, this is an issue only on Windows. AutoInterface will work as you expect on any other system.

> Ok, that is clear. Thanks for advice.But what will be if mentioned PC will be temporary not connected, but another PC(laptop) will connect to same network?

If the only Windows machine goes offline, everything else will keep using AutoInterface. If you have a number of Windows machines, then there will need to be at least one server at any time. 

If you can do so, if you can get a Linux system, like a Raspberry Pi, then it can communicate both on the AutoInterface and run a server for all the Windows systems that might join or leave. I realize being at sea makes that difficult, but it's something to consider the next time you're able to order one.

**Reply by faragher** [reply]
> Sorry for being annoying....

Don't worry! This is an important thing to set up, since there are a lot of people who could benefit from a network on a ship.

**Reply by mrmate2** [reply]
Dear farager, many thanks for Your replies and time spent. now a lot of things are clear for me!
